### ISSUE TITLE
[clang-tidy] readability-identifier-naming False positive and destructive --fix on templated 'using' aliases

### ISSUE THREAD

**Issue by MichaelJacobsAuthentrics** [llvm/llvm-project#215396]
Tested on clang-tidy 22.1.8. Also tested on clang-tidy 20 with no false positive.

Minimal example:
```C++
// main.cpp

#include <vector>

template <typename T>
using AliasType = std::vector<T>;

int main()
{
    return 0;
}
```

```sh
$ clang-tidy --checks='-*,readability-identifier-naming' main.cpp
1 warning generated.
repro.cpp:8:1: warning: invalid case style for identifier 'AliasType' [readability-identifier-naming]
    8 | using AliasType = std::vector<T>;
      | ^~~~~
      | alias_type
```

**Comment by vbvictor** [comment]
The issue seem to no longer reproduce on trunk, please try most recent release (23) or main
https://godbolt.org/z/svrWP5YhM

**Comment by vbvictor** [comment]
If this still exists on 23/main, please post a reproducer (preferrably godbolt) and reopen the issue
